server/driver/wivrn_controller.cpp

```diff
@@ -811,6 +818,17 @@ void wivrn_controller::set_derived_pose(const from_headset::derived_pose & deriv
 
 void wivrn_controller::update_tracking(const from_headset::tracking & tracking, const clock_offset & offset)
 {
+	const bool left = device_type == XRT_DEVICE_TYPE_LEFT_HAND_CONTROLLER;
+
+	xrt_system_roles roles;
+	cnx->get_roles(&roles);
+	if (cnx->get_xdev(left ? roles.left : roles.right) != this)
+	{
+		// we don't need to disable aim/grip/palm, hand_interaction will
+		// end up disabling those when unused
+		return;
```

Comment:
The first update_tracking after a change will disable them, because the last stored request will be too old for the new data.

Also, the get_roles/get_xdev dance is more complicated than necessary.
there's already some dispatch logic in wivrn_session already, in the end it may be better to have all of them in wivrn_session.

The alternative would be for the controller/hand_interaction object to know if it's active and move all the discard logic there, the current split is awkward.